torchtune/_cli/download.py

```diff
@@ -4,70 +4,101 @@
+                Successfully downloaded model repo and wrote to the following locations:
+                ./model/config.json
+                ./model/README.md
+                ./model/consolidated.00.pth
+                ...
 
-    Raises:
-        ValueError: If the model is not supported.
-    """
-    if "meta-llama" in repo_id and hf_token is None:
-        raise ValueError(
-            "You need to provide a Hugging Face API token to download gated models."
-            "You can find your token by visiting https://huggingface.co/settings/tokens"
-        )
+                # Download an ungated model from the HuggingFace Hub
+                $ tune download mistralai/Mistral-7B-Instruct-v0.2
+                Successfully downloaded model repo and wrote to the following locations:
+                ./model/config.json
+                ./model/README.md
+                ./model/model-00001-of-00003.safetensors
+                ...
 
-    # Download the tokenizer and PyTorch model files
-    snapshot_download(
-        repo_id,
-        local_dir=output_dir,
-        resume_download=True,
-        token=hf_token,
-    )
+            For a list of all models, visit the HuggingFace Hub https://huggingface.co/models.
+            """
+            ),
+            formatter_class=argparse.RawTextHelpFormatter,
+        )
+        self._add_arguments()
+        self._parser.set_defaults(func=self._download_cmd)
 
-    print(
-        "Succesfully downloaded model repo and wrote to the following locations:",
-        *list(output_dir.iterdir()),
-        sep="\n",
-    )
+    def _add_arguments(self) -> None:
+        """Add arguments to the parser."""
+        self._parser.add_argument(
+            "repo_id",
+            type=str,
+            help="Name of the repository on HuggingFace Hub.",
+        )
+        self._parser.add_argument(
+            "--output-dir",
+            type=Path,
+            required=False,
+            default="./model",
+            help="Directory in which to save the model.",
+        )
+        self._parser.add_argument(
+            "--hf-token",
+            type=str,
+            required=False,
+            default=os.getenv("HF_TOKEN", None),
+            help="HuggingFace API token. Needed for gated models like Llama2.",
+        )
 
+    def _download_cmd(self, args: argparse.Namespace) -> None:
+        """Downloads a model from the HuggingFace Hub."""
+        # Download the tokenizer and PyTorch model files
+        try:
+            true_output_dir = snapshot_download(
+                args.repo_id,
+                local_dir=args.output_dir,
+                resume_download=True,
+                token=args.hf_token,
+            )
+        except GatedRepoError:
+            self._parser.error(
+                "You need to provide a HuggingFace API token to download gated models."
+                "You can find your token by visiting https://huggingface.co/settings/tokens"
```

Comment:
Is this the only reason that download from a gated repo can fail? What if the user has not signed the consent form or something like that? In that case is this error misleading?

Reply:
Yeah, I can update this comment to be more descriptive.